Objects/funcobject.c

```diff
@@ -134,6 +134,9 @@ uint32_t _PyFunction_GetVersionForCurrentState(PyFunctionObject *func)
     if (func->func_version != 0) {
         return func->func_version;
     }
+    if (func->vectorcall != _PyFunction_Vectorcall) {
+        return 0;
```

Comment:
@markshannon @brandtbucher @gvanrossum: I don't know if this change is correct, I don't know how _PyFunction_GetVersionForCurrentState() is supposed to work.

Reply:
This is correct. `_PyFunction_GetVersionForCurrentState` returns 0 if it cannot set the version number to a valid version.